order/src/main/java/com/yas/order/service/CartItemService.java

```diff
@@ -32,13 +37,32 @@ public void addCartItem(CartItemPostVm cartItemPostVm) {
         }
     }
 
+    public CartItemGetVm updateCartItem(Long productId, CartItemPutVm cartItemPutVm) {
+        validateProduct(productId);
+
+        String currentUserId = AuthenticationUtils.getCurrentUserId();
+        CartItemId cartItemId = CartItemId.of(currentUserId, productId);
+
+        Optional<CartItem> cartItemOpt = cartItemRepository.findById(cartItemId);
+
+        CartItem cartItem;
+        if (cartItemOpt.isPresent()) {
+            cartItem = cartItemOpt.get();
+            cartItem.setQuantity(cartItemPutVm.quantity());
+        } else {
+            cartItem = CartItem.builder()
+                .customerId(currentUserId)
+                .productId(productId)
+                .quantity(cartItemPutVm.quantity())
+                .build();
+        }
```

Comment:
Should avoid re-assigned local variable & create a method inside mapper to create newCartItem, suggest change:
```java
CartItem cartItem = cartItemRepository.findById(cartItemId)
    .map(existingCartItem -> {
        existingCartItem.setQuantity(cartItemPutVm.quantity());
        return existingCartItem;
    })
    .orElseGet(() -> cartItemMapper.newCartItem(
        currentUserId, 
        productId,
        cartItemPutVm.quantity()
    ));
```

Reply:
Hi @caominhduc52, I've pushed the changes. I'd appreciate it if you could take a look. Thank you.